docs_src/design/adr/device-service/0012-DeviceService-Filters.md

```diff
@@ -0,0 +1,152 @@
+
+``` go
+func (f Filter) FilterByValueInRange (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+
+func (f Filter) FilterByValueOutRange (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+
+func (f Filter) FilterByValueEqual (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+
+func (f Filter) FilterByValueNotEqual (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+
+func (f Filter) FilterByResourceNamesMatch (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+```
+
+!!! Note
+    The app functions SDK comes with `FilterByDeviceName` and `FilterByResourceName` functions today. The FilterByResourceName would behave similarly to FilterByResourceNameMatch.
+
+    The Filter structure houses the configuration parameters for which the filter functions work and filter on.
+
+A NewFilter function creates, initializes and returns a new instance of the filter based on the configuration provided.
+
+``` go
+func NewReadingNameFilter(filterValues []string) Filter {
+    return Filter{FilterValues: filterValues}
+}
+```
+
+### Sharing filter functions
+
+If one were to explore the filtering functions in the app functions SDK [filter.go](https://github.com/edgexfoundry/app-functions-sdk-go/blob/master/pkg/transforms/filter.go) (both `FilterByDeviceName` and `FilterByValueDescriptor`), the filters operate on the `Event` model object and return the same objects (`Event` or nil).  Ideally, since both app services and device services generally share the same interface model (from `go-mod-core-contracts`), it would be the desire to share the same filter functions functions between SDKs and associated services.
+
+Decisions on how to do this in Go - whether by shared module for example - is left as an implementation detail.  C needs are likely to be handled in the SDK directly.
+
+#### Additional Design Considerations
+
+As Device Services do not have the concept of a functions pipeline like application services do, consideration must be given as to how and where to:
+
+- provide configuration to specify which filter functions to invoke
+- create the filter
+- invoke the filtering functions
+
+At this time, custom filters will not be supported as the custom filters would not be known by the SDK and therefore could not be specified in configuration.  This is consistent with the app functions SDK and filtering.
+
+#### Function Inflection Point
+
+When instructed to "get" new readings, the function [`execReadDeviceResource`](https://github.com/edgexfoundry/device-sdk-go/blob/0bbcb663a9153978e7e9ef8c297d5988e58906d0/internal/handler/command.go) is called which subsequently calls on the the device service driver's `HandleReadCommands` function to get the latest sensor values from the device.  The driver's `HandleReadCommands` returns the sensor reading data (via array of CommandValues) to be put into an event.
```

Comment:
I think this is too much detail, especially given the re-factoring being done for V2. I think it would be simpler to just state that filtering always comes after `Event` composition, and therefore results in filtered results being made available prior to REST replies being sent and/or , and `Events` being pushed to Core Data or the message bus.